File: insta-lab4/instagram/app/src/main/java/com/example/instagram/DetailsActivity.java
package com.example.instagram;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Html;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

public class DetailsActivity extends AppCompatActivity {

    ImageButton back;
    ImageView likeBtn;
    ImageView saveBtn;
    ImageView profilePhoto;
    TextView author;
    ImageView postImage;
    TextView postText;
    TextView date;
    TextView likesCnt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news_detail);

        back = findViewById(R.id.back);
        likeBtn = findViewById(R.id.likeBtn);
        saveBtn = findViewById(R.id.saveBtn);
        profilePhoto = findViewById(R.id.profilePhoto);
        author = findViewById(R.id.author);
        postImage = findViewById(R.id.postImage);
        postText = findViewById(R.id.postText);
        date = findViewById(R.id.date);
        likesCnt = findViewById(R.id.likesCnt);

        final News news = (News)getIntent().getSerializableExtra("news");

        profilePhoto.setImageResource(news.getProfilePhoto());
        postImage.setImageResource(news.getPostImage());

        author.setText(news.getAuthor());
        String s = "<b>"+ news.getAuthor() + "</b>" + " " + news.getPostText();
        postText.setText(Html.fromHtml(s));
        date.setText(news.getDate());
        likesCnt.setText(news.getLikesCnt()+" likes");

        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
        likeBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (news.getLikeBtn()==R.drawable.like){
                    likeBtn.setImageResource(R.drawable.liked);
                    news.setLikeBtn(R.drawable.liked);
                } else {
                    likeBtn.setImageResource(R.drawable.like);
                    news.setLikeBtn(R.drawable.like);
                }
            }
        });
    }
}

File: SocialNetKotlinePart2/app/src/main/java/com/example/socialnetwork/view/DetailsActivity.kt
package com.example.socialnetwork.view

import android.os.Bundle
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.example.socialnetwork.R
import com.example.socialnetwork.model.PostData
import com.example.socialnetwork.view_model.PostViewModel
import kotlin.properties.Delegates

class DetailsActivity : AppCompatActivity() {

    private lateinit var postLogo: ImageView
    private lateinit var postImage: ImageView
    private lateinit var postAuthor: TextView
    private lateinit var postDescription: TextView
    private lateinit var postLike: TextView
    private lateinit var postComments: TextView
    private lateinit var postTime: TextView
    private lateinit var backButton: ImageButton
    private var postId by Delegates.notNull<Int>()
    private val postViewModel: PostViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_details)
        postId = intent.getIntExtra("post_id", 1)
        bindView()
        setData()
    }

    private fun bindView() {
        postLogo = findViewById(R.id.postLogo)
        postImage = findViewById(R.id.postImage)
        postAuthor = findViewById(R.id.postAuthorTextView)
        postDescription = findViewById(R.id.postDescriptionTextView)
        postLike = findViewById(R.id.postLikeTextView)
        postComments = findViewById(R.id.postCommentsTextView)
        postTime = findViewById(R.id.postTimeTextView)
        backButton = findViewById(R.id.backImageButton)

        backButton.setOnClickListener {
            onBackPressed()
        }
    }

    private fun setData() {
        postViewModel.searchPost(postId = postId)
        postViewModel.liveData.observe(this, Observer { result ->
            when (result) {
                is PostViewModel.State.SearchResult -> {
                    setView(result.post)
                }
            }
        })
    }

    private fun setView(postData: PostData) {
        postLogo.setImageResource(postData.postLogo)
        postImage.setImageResource(postData.postImage)
        postAuthor.text = postData.postAuthor
        postLike.text = "Нравиться: " + postData.postLike.toString()
        postDescription.text = postAuthor.text as String + " " + postData.postDescription
        postComments.text = postData.postComments
        postTime.text = postData.postTime
    }

    override fun onBackPressed() {
        super.onBackPressed()
        return
    }
}